interfaces/builtin/jack.go

```diff
@@ -0,0 +1,52 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const jackSummary = `allows access to the jack socket`
+
+const jackBaseDeclarationSlots = `
+  jack:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+/*
+ * libjack builds the shared memory endpoint like this :
+ * jack-<userid>/<server name>/jack-<server id>
+ *
+ * see libjack/shm.c in jack1's source tree,
+ * and common/shm.c in jack2's source tree.
+ */
+const jackConnectedPlugAppArmor = `
+/dev/shm/jack-*/*/* mrwk,
```

Comment:
Not a full review, but I believe this can be changed to:
```
owner /dev/shm/jack-[0-9]*/*/* mrwk,
```

but I also wonder why you need 'm' and 'k'. What are the policy violations you see in devmode when this interface is not connected?

Do you have a test snap and instructions on how to setup jack so that I may poke at this a little bit?

Reply:
To be honest I took thise from the pulse interface. I'm happy to change that and retest, then I'll update the PR. 

As for the test snap and the configuration for jackd, I believe you have everything necessary in the PR description. Let me know if you need anything else.

Reply:
I've changed the profile as per your suggestions.
I've also updated [my test snap](https://github.com/frenchguych/jack-test) so it now plays a 880 Hz tone @ 60 bpm. Everything sounds good on my machine.